apps/settings/js/modules/dialog_manager.js

```diff
@@ -177,12 +177,21 @@ define(function(require) {
         return self._getPanel(dialog.panel.id);
       })
       .then(function(panel) {
-        // 4: call beforeShow
+        // 4: Inject cancel(), submit() functions for panel use
+        if (!panel.__injectNeededFunctions) {
```

Comment:
I would suggest not to attach the functions directly to `panel` but defining a proper interface for assigning the functions in `dialog_panel.js`.